website/pages/docs/secrets/key-management/index.mdx

```diff
@@ -0,0 +1,189 @@
+    ```
+
+    Conceptually, a KMS provider resource represents a destination for keys to be distributed to
+    and subsequently managed in. It is configured using a generic set of parameters. The values
+    supplied to the generic set of parameters will differ depending on the specified `provider`.
+
+    This operation creates a KMS provider that represents a named Azure Key Vault instance.
+    This is accomplished by specifying the `azurekeyvault` provider along with other provider-specific
+    parameter values. For details on how to configure each supported KMS provider, see the
+    [KMS Providers](#kms-providers) section.
+
+1. Distribute a key to a KMS provider:
+
+    ```text
+    $ vault write keymgmt/kms/example-kms/key/example-key \
+        purpose="encrypt,decrypt" \
+        protection="hsm"
+    ```
+
+   This operation distributes a **copy** of the named key to the KMS provider with a specific
+   `purpose` and `protection`. The `purpose` defines the set of cryptographic capabilities
+   that the key will have in the KMS provider. The `protection` defines where cryptographic
+   operations are performed with the key in the KMS provider. See the API documentation for a list of
+   supported [purpose](/api/secret/key-management#purpose) and [protection](/api/secret/key-management#protection)
+   values.
+
+   ~> The amount of time it takes to distribute a key to a KMS provider is proportional to the
+   number of versions that the key has. If a timeout occurs when distributing a key to a KMS
+   provider, you may need to increase the [VAULT_CLIENT_TIMEOUT](https://www.vaultproject.io/docs/commands#vault_client_timeout).
+
+1. Rotate a key:
+
+    ```text
+    $ vault write -f keymgmt/key/example-key/rotate
+    ```
+
+   Rotating a key creates a new key version that contains new key material. The key will be rotated
+   in both Vault and the KMS provider that the key has been distributed to. The new key version
+   will be enabled and set as the current version for cryptographic operations in the KMS provider.
+
+1. Enable or disable key versions:
+
+    ```text
+    $ vault write keymgmt/key/example-key min_enabled_version=2
+    ```
+
+    The `min_enabled_version` of a key can be updated in order to enable or disable sequences of
+    key versions. All versions of the key less than the `min_enabled_version` will be disabled for
+    cryptographic operations in the KMS provider that the key has been distributed to. Setting a
+    `min_enabled_version` of `0` means that all key versions will be enabled.
+
+1. Remove a key from a KMS provider:
+
+    ```text
+    $ vault delete keymgmt/kms/example-kms/key/example-key
+    ```
+
+    This operation results in the key being deleted from the KMS provider. The key will still exist
+    in the secrets engine and can be redistributed to a KMS provider at a later time.
+
+    To permanently delete the key from the secrets engine, the [delete key](/api/secret/key-management#delete-key)
```

Comment:
Since it's a `DELETE` operation, I believe this can be done via `vault delete ...`

Reply:
This can be done using the CLI. I didn't add a step to the usage section for this because it's a simple operation that doesn't need much conceptual explaining. I thought it was pretty common knowledge that the CLI is mostly just an API client, so I refer to the API here. Happy to add another step with CLI usage if you think that's better.